### PR TITLE
parity with js version

### PR DESCRIPTION
**Pull request by morishuz**
- added important changes from new js version to cpp
- removed issue with std::move. coords is now just an argument: simpler and safer. 
- simplified sorting (helped with previous point)
- added tests to compare 5000 vertex triangulation with new js version

### CODE REVIEW

File: src/delaunator.cpp [hunk 1]
```diff
@@ -229,76 +227,73 @@ Delaunator::Delaunator(vector<double> in_coords) {
...
-                m_hull[m_hull[e].next].x, m_hull[m_hull[e].next].y
-            ) >= 0
-        ) {
-            e = m_hull[e].next;
+        while(!orient(x, y,m_hl[e].x, m_hl[e].y,m_hl[m_hl[e].next].x, m_hl[m_hl[e].next].y))
+        {
+            e = m_hl[e].next;
 
             if (e == start) {
                 throw runtime_error("Something is wrong with the input points.");
```
Comment (mourner) [inline src/delaunator.cpp]: This place is different in JS — let's port over the fix here too.

File: src/delaunator.cpp [hunk 2]
```diff
@@ -136,36 +129,43 @@ namespace {
     }
 }
 
-Delaunator::Delaunator(vector<double> in_coords) {
-    coords = move(in_coords);
-    const long int n = coords.size() >> 1;
+Delaunator::Delaunator()
+{
+    m_epilon = std::pow(2,-52);
```
Comment (mourner) [inline src/delaunator.cpp]: typo (epsilon)

File: src/delaunator.cpp (hunk 1 above)
Comment (morishuz) [reply src/delaunator.cpp]: very good, i missed that. fixed now

**Comment by delfrrr** [comment]
@morishuz thank you for PR! Unfortunately I have to suggest to close it:

1. Please note #3 and #6 ; it will change literally all files and complicated conflicts are inevitable. I suggest to continue with improvements after #6 is merged (or on top of #6 )
2. Please, avoid big multi-feature PR (this one can easily split on 4 PR's)
3. Before implementing parity with JS version code we need parity with JS version tests

**Comment by delfrrr** [comment]
Already fixed by other commits, see https://github.com/delfrrr/delaunator-cpp/issues/2

**Pull request closed without merging**